.github/workflows/deploy.yml

```
@@ -2,25 +1,25 @@
 
 jobs:
   build-and-push:
-    if: ${{ github.event.pull_request.merged || contains(github.event.head_commit.message, '[deploy]') }} # Only run for PR merge or push with "[deploy]"
     runs-on: ubuntu-latest
+    if: contains(github.event.head_commit.message, '[deploy]') # Condition to check the commit message
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
```

actionlint (1.7.4)

the runner of "docker/setup-buildx-action@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)